Minio/ApiEndpoints/BucketOperations.cs

```diff
@@ -441,5 +448,130 @@ public IObservable<MinioNotificationRaw> ListenBucketNotificationsAsync(string b
+                }
+                var request = await this.CreateRequest(Method.GET, bucketName).ConfigureAwait(false);
+                request.AddQueryParameter("versioning","");
+                var response = await this.ExecuteTaskAsync(this.NoErrorHandlers, request, cancellationToken).ConfigureAwait(false);
+                var contentBytes = System.Text.Encoding.UTF8.GetBytes(response.Content);
+                VersioningConfiguration config = null;
+                if (HttpStatusCode.OK.Equals(response.StatusCode))
+                {
+                    using (var stream = new MemoryStream(contentBytes))
+                    {
+                        config = (VersioningConfiguration)new XmlSerializer(typeof(VersioningConfiguration)).Deserialize(stream);
+                    }
+                    return config.Status.Equals("Enabled");
+                }
+                else
+                {
+                    return false;
+                }
+            }
+            catch (Exception ex)
+            {
+                if (ex.GetType() == typeof(BucketNotFoundException))
+                {
+                    return false;
+                }
+                throw;
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// Enable versioning on a bucket
+        /// </summary>
+        /// <param name="bucketName">Name of the new bucket</param>
+        /// <param name="cancellationToken">Optional cancellation token to cancel the operation</param>
+        /// <returns>Task that returns true if versioning is enabled on the bucket</returns>
+        public async Task EnableVersioningAsync(string bucketName, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            // Set Target URL
+            try
+            {
+                if (bucketName == null)
+                {
+                    throw new InvalidBucketNameException(bucketName, "bucketName cannot be null");
+                }
+                Uri requestUrl = RequestUtil.MakeTargetURL(this.BaseUrl, this.Secure, bucketName);
+                SetTargetURL(requestUrl);
+                VersioningConfiguration config = new VersioningConfiguration(true);
+                var request = new RestRequest("/" + bucketName, Method.PUT)
+                {
+                    XmlSerializer = new RestSharp.Serializers.DotNetXmlSerializer()
+                    {
+                        Namespace = "http://s3.amazonaws.com/doc/2006-03-01/",
+                        ContentType = config.ToString()
+                    },
+                    RequestFormat = DataFormat.Xml
+                };
+                string body = Regex.Replace(request.XmlSerializer.Serialize(config), @"\t|\n|\r", "");
+                string replace1 = ">\\s+";
+                string replace2 = "\\s+<";
+                body = Regex.Replace(body, replace1, ">");
+                body = Regex.Replace(body, replace2, "<");
```

Comment:
@BigUstad , you don't have to do a regex replacement. Just need to marshal the xml and UTF-8 encode. See how SelectObjectContentAsync request is being marshalled as an example.

Reply:
@poornas I did make the change. Thanks for pointing out.